mbed_host_tests/host_tests_runner/host_test_default.py

```diff
@@ -117,7 +117,7 @@ def callback__notify_prn(key, value, timestamp):
             consume_preamble_events = True
             while (time() - start_time) < timeout_duration:
                 # Handle default events like timeout, host_test_name, ...
-                if event_queue.qsize():
+                if not event_queue.empty():
```

Comment:
While ```not event_queue.empty():``` yields False we would be in a tight loop in ```while``` on line #118. 
This if statement is not required as execution of this loop depends on data coming in the ```event_queue```.

Implementation can look like:
```
while True:
    rem = timeout_duration - (time() - start_time)
    if rem > 0: 
        break

    try:
        (key, value, timestamp) = event_queue.get(timeout=rem)
    except QueueEmpty:
        continue
```